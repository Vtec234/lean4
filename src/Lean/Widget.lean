import Lean.Widget.ToHtmlFormat
import Lean.Widget.PrintableExpr
import Lean.Widget.WidgetGoal
import Lean.Widget.WidgetMessage
