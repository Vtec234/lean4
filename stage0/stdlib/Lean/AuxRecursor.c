// Lean compiler output
// Module: Lean.AuxRecursor
// Imports: Init Lean.Environment
#include <lean/lean.h>
#if defined(__clang__)
#pragma clang diagnostic ignored "-Wunused-parameter"
#pragma clang diagnostic ignored "-Wunused-label"
#elif defined(__GNUC__) && !defined(__CLANG__)
#pragma GCC diagnostic ignored "-Wunused-parameter"
#pragma GCC diagnostic ignored "-Wunused-label"
#pragma GCC diagnostic ignored "-Wunused-but-set-variable"
#endif
#ifdef __cplusplus
extern "C" {
#endif
lean_object* l_Lean_mkRecOnName(lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__4___boxed(lean_object*, lean_object*);
lean_object* lean_mk_empty_array_with_capacity(lean_object*);
lean_object* lean_mark_aux_recursor(lean_object*, lean_object*);
lean_object* lean_name_mk_string(lean_object*, lean_object*);
lean_object* l_Lean_noConfusionExt;
lean_object* l_Lean_noConfusionExt___elambda__2___boxed(lean_object*);
uint8_t l_Lean_isCasesOnRecursor(lean_object*, lean_object*);
lean_object* l_Lean_auxRecExt___elambda__1___boxed(lean_object*);
lean_object* l_Lean_auxRecExt___elambda__4(lean_object*, lean_object*);
lean_object* l_Lean_auxRecExt;
lean_object* l_Lean_binductionOnSuffix;
lean_object* l_Lean_mkCasesOnName(lean_object*);
lean_object* l_Lean_auxRecExt___elambda__4___rarg(lean_object*);
lean_object* l_Lean_recOnSuffix;
lean_object* l_Lean_isNoConfusion___boxed(lean_object*, lean_object*);
lean_object* lean_mark_no_confusion(lean_object*, lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__3___boxed(lean_object*, lean_object*);
lean_object* l_Lean_auxRecExt___elambda__1(lean_object*);
lean_object* l_Lean_initFn____x40_Lean_AuxRecursor___hyg_39_(lean_object*);
lean_object* l_Lean_initFn____x40_Lean_AuxRecursor___hyg_101_(lean_object*);
lean_object* l_Lean_isCasesOnRecursor___boxed(lean_object*, lean_object*);
lean_object* l_Lean_isCasesOnRecursor_match__1___rarg(lean_object*, lean_object*, lean_object*);
lean_object* l_Lean_auxRecExt___elambda__3(lean_object*, lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__3(lean_object*, lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__1___boxed(lean_object*);
lean_object* l_Lean_PersistentEnvExtension_addEntry___rarg(lean_object*, lean_object*, lean_object*);
lean_object* l_Lean_EnvExtensionInterfaceUnsafe_instInhabitedExt___lambda__1(lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__4___rarg(lean_object*);
lean_object* l_Lean_auxRecExt___elambda__2___boxed(lean_object*);
uint8_t lean_is_no_confusion(lean_object*, lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__1(lean_object*);
lean_object* l_Lean_auxRecExt___elambda__3___boxed(lean_object*, lean_object*);
uint8_t lean_is_aux_recursor(lean_object*, lean_object*);
lean_object* l_Lean_auxRecExt___elambda__2(lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__4(lean_object*, lean_object*);
lean_object* l_Lean_mkBRecOnName(lean_object*);
lean_object* l_Lean_mkTagDeclarationExtension(lean_object*, lean_object*);
lean_object* l_Lean_isCasesOnRecursor_match__1(lean_object*);
uint8_t l_Lean_TagDeclarationExtension_isTagged(lean_object*, lean_object*, lean_object*);
lean_object* l_Lean_isAuxRecursor___boxed(lean_object*, lean_object*);
lean_object* l_Lean_brecOnSuffix;
lean_object* l_Lean_auxRecExt___elambda__4___boxed(lean_object*, lean_object*);
uint32_t lean_uint32_of_nat(lean_object*);
lean_object* l_Lean_noConfusionExt___elambda__2(lean_object*);
lean_object* l_Lean_casesOnSuffix;
lean_object* l_Lean_mkBInductionOnName(lean_object*);
uint8_t lean_string_dec_eq(lean_object*, lean_object*);
static lean_object* _init_l_Lean_casesOnSuffix() {
_start:
{
lean_object* x_1; 
x_1 = lean_mk_string("casesOn");
return x_1;
}
}
static lean_object* _init_l_Lean_recOnSuffix() {
_start:
{
lean_object* x_1; 
x_1 = lean_mk_string("recOn");
return x_1;
}
}
static lean_object* _init_l_Lean_brecOnSuffix() {
_start:
{
lean_object* x_1; 
x_1 = lean_mk_string("brecOn");
return x_1;
}
}
static lean_object* _init_l_Lean_binductionOnSuffix() {
_start:
{
lean_object* x_1; 
x_1 = lean_mk_string("binductionOn");
return x_1;
}
}
lean_object* l_Lean_mkCasesOnName(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; 
x_2 = l_Lean_casesOnSuffix;
x_3 = lean_name_mk_string(x_1, x_2);
return x_3;
}
}
lean_object* l_Lean_mkRecOnName(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; 
x_2 = l_Lean_recOnSuffix;
x_3 = lean_name_mk_string(x_1, x_2);
return x_3;
}
}
lean_object* l_Lean_mkBRecOnName(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; 
x_2 = l_Lean_brecOnSuffix;
x_3 = lean_name_mk_string(x_1, x_2);
return x_3;
}
}
lean_object* l_Lean_mkBInductionOnName(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; 
x_2 = l_Lean_binductionOnSuffix;
x_3 = lean_name_mk_string(x_1, x_2);
return x_3;
}
}
lean_object* l_Lean_initFn____x40_Lean_AuxRecursor___hyg_39_(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; lean_object* x_4; lean_object* x_5; 
x_2 = lean_box(0);
x_3 = lean_mk_string("auxRec");
x_4 = lean_name_mk_string(x_2, x_3);
x_5 = l_Lean_mkTagDeclarationExtension(x_4, x_1);
return x_5;
}
}
lean_object* l_Lean_auxRecExt___elambda__1(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = lean_box(0);
return x_2;
}
}
lean_object* l_Lean_auxRecExt___elambda__2(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; 
x_2 = lean_unsigned_to_nat(0u);
x_3 = lean_mk_empty_array_with_capacity(x_2);
return x_3;
}
}
lean_object* l_Lean_auxRecExt___elambda__3(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_inc(x_1);
return x_1;
}
}
lean_object* l_Lean_auxRecExt___elambda__4___rarg(lean_object* x_1) {
_start:
{
lean_object* x_2; uint32_t x_3; lean_object* x_4; lean_object* x_5; lean_object* x_6; 
x_2 = lean_unsigned_to_nat(0u);
x_3 = lean_uint32_of_nat(x_2);
x_4 = lean_mk_string("");
x_5 = lean_alloc_ctor(0, 1, 4);
lean_ctor_set(x_5, 0, x_4);
lean_ctor_set_uint32(x_5, sizeof(void*)*1, x_3);
x_6 = lean_alloc_ctor(1, 2, 0);
lean_ctor_set(x_6, 0, x_5);
lean_ctor_set(x_6, 1, x_1);
return x_6;
}
}
lean_object* l_Lean_auxRecExt___elambda__4(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; 
x_3 = lean_alloc_closure((void*)(l_Lean_auxRecExt___elambda__4___rarg), 1, 0);
return x_3;
}
}
lean_object* l_Lean_auxRecExt___elambda__1___boxed(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = l_Lean_auxRecExt___elambda__1(x_1);
lean_dec(x_1);
return x_2;
}
}
lean_object* l_Lean_auxRecExt___elambda__2___boxed(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = l_Lean_auxRecExt___elambda__2(x_1);
lean_dec(x_1);
return x_2;
}
}
lean_object* l_Lean_auxRecExt___elambda__3___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; 
x_3 = l_Lean_auxRecExt___elambda__3(x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
return x_3;
}
}
lean_object* l_Lean_auxRecExt___elambda__4___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; 
x_3 = l_Lean_auxRecExt___elambda__4(x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
return x_3;
}
}
lean_object* lean_mark_aux_recursor(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; lean_object* x_4; 
x_3 = l_Lean_auxRecExt;
x_4 = l_Lean_PersistentEnvExtension_addEntry___rarg(x_3, x_1, x_2);
return x_4;
}
}
uint8_t lean_is_aux_recursor(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; uint8_t x_4; 
x_3 = l_Lean_auxRecExt;
x_4 = l_Lean_TagDeclarationExtension_isTagged(x_3, x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
return x_4;
}
}
lean_object* l_Lean_isAuxRecursor___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
uint8_t x_3; lean_object* x_4; 
x_3 = lean_is_aux_recursor(x_1, x_2);
x_4 = lean_box(x_3);
return x_4;
}
}
lean_object* l_Lean_isCasesOnRecursor_match__1___rarg(lean_object* x_1, lean_object* x_2, lean_object* x_3) {
_start:
{
if (lean_obj_tag(x_1) == 1)
{
lean_object* x_4; lean_object* x_5; uint64_t x_6; lean_object* x_7; lean_object* x_8; 
lean_dec(x_3);
x_4 = lean_ctor_get(x_1, 0);
lean_inc(x_4);
x_5 = lean_ctor_get(x_1, 1);
lean_inc(x_5);
x_6 = lean_ctor_get_uint64(x_1, sizeof(void*)*2);
lean_dec(x_1);
x_7 = lean_box_uint64(x_6);
x_8 = lean_apply_3(x_2, x_4, x_5, x_7);
return x_8;
}
else
{
lean_object* x_9; 
lean_dec(x_2);
x_9 = lean_apply_1(x_3, x_1);
return x_9;
}
}
}
lean_object* l_Lean_isCasesOnRecursor_match__1(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = lean_alloc_closure((void*)(l_Lean_isCasesOnRecursor_match__1___rarg), 3, 0);
return x_2;
}
}
uint8_t l_Lean_isCasesOnRecursor(lean_object* x_1, lean_object* x_2) {
_start:
{
if (lean_obj_tag(x_2) == 1)
{
lean_object* x_3; lean_object* x_4; uint8_t x_5; 
x_3 = lean_ctor_get(x_2, 1);
x_4 = l_Lean_casesOnSuffix;
x_5 = lean_string_dec_eq(x_3, x_4);
if (x_5 == 0)
{
uint8_t x_6; 
x_6 = 0;
return x_6;
}
else
{
lean_object* x_7; uint8_t x_8; 
x_7 = l_Lean_auxRecExt;
x_8 = l_Lean_TagDeclarationExtension_isTagged(x_7, x_1, x_2);
return x_8;
}
}
else
{
uint8_t x_9; 
x_9 = 0;
return x_9;
}
}
}
lean_object* l_Lean_isCasesOnRecursor___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
uint8_t x_3; lean_object* x_4; 
x_3 = l_Lean_isCasesOnRecursor(x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
x_4 = lean_box(x_3);
return x_4;
}
}
lean_object* l_Lean_initFn____x40_Lean_AuxRecursor___hyg_101_(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; lean_object* x_4; lean_object* x_5; 
x_2 = lean_box(0);
x_3 = lean_mk_string("noConf");
x_4 = lean_name_mk_string(x_2, x_3);
x_5 = l_Lean_mkTagDeclarationExtension(x_4, x_1);
return x_5;
}
}
lean_object* l_Lean_noConfusionExt___elambda__1(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = lean_box(0);
return x_2;
}
}
lean_object* l_Lean_noConfusionExt___elambda__2(lean_object* x_1) {
_start:
{
lean_object* x_2; lean_object* x_3; 
x_2 = lean_unsigned_to_nat(0u);
x_3 = lean_mk_empty_array_with_capacity(x_2);
return x_3;
}
}
lean_object* l_Lean_noConfusionExt___elambda__3(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_inc(x_1);
return x_1;
}
}
lean_object* l_Lean_noConfusionExt___elambda__4___rarg(lean_object* x_1) {
_start:
{
lean_object* x_2; uint32_t x_3; lean_object* x_4; lean_object* x_5; lean_object* x_6; 
x_2 = lean_unsigned_to_nat(0u);
x_3 = lean_uint32_of_nat(x_2);
x_4 = lean_mk_string("");
x_5 = lean_alloc_ctor(0, 1, 4);
lean_ctor_set(x_5, 0, x_4);
lean_ctor_set_uint32(x_5, sizeof(void*)*1, x_3);
x_6 = lean_alloc_ctor(1, 2, 0);
lean_ctor_set(x_6, 0, x_5);
lean_ctor_set(x_6, 1, x_1);
return x_6;
}
}
lean_object* l_Lean_noConfusionExt___elambda__4(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; 
x_3 = lean_alloc_closure((void*)(l_Lean_noConfusionExt___elambda__4___rarg), 1, 0);
return x_3;
}
}
lean_object* l_Lean_noConfusionExt___elambda__1___boxed(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = l_Lean_noConfusionExt___elambda__1(x_1);
lean_dec(x_1);
return x_2;
}
}
lean_object* l_Lean_noConfusionExt___elambda__2___boxed(lean_object* x_1) {
_start:
{
lean_object* x_2; 
x_2 = l_Lean_noConfusionExt___elambda__2(x_1);
lean_dec(x_1);
return x_2;
}
}
lean_object* l_Lean_noConfusionExt___elambda__3___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; 
x_3 = l_Lean_noConfusionExt___elambda__3(x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
return x_3;
}
}
lean_object* l_Lean_noConfusionExt___elambda__4___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; 
x_3 = l_Lean_noConfusionExt___elambda__4(x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
return x_3;
}
}
lean_object* lean_mark_no_confusion(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; lean_object* x_4; 
x_3 = l_Lean_noConfusionExt;
x_4 = l_Lean_PersistentEnvExtension_addEntry___rarg(x_3, x_1, x_2);
return x_4;
}
}
uint8_t lean_is_no_confusion(lean_object* x_1, lean_object* x_2) {
_start:
{
lean_object* x_3; uint8_t x_4; 
x_3 = l_Lean_noConfusionExt;
x_4 = l_Lean_TagDeclarationExtension_isTagged(x_3, x_1, x_2);
lean_dec(x_2);
lean_dec(x_1);
return x_4;
}
}
lean_object* l_Lean_isNoConfusion___boxed(lean_object* x_1, lean_object* x_2) {
_start:
{
uint8_t x_3; lean_object* x_4; 
x_3 = lean_is_no_confusion(x_1, x_2);
x_4 = lean_box(x_3);
return x_4;
}
}
lean_object* initialize_Init(lean_object*);
lean_object* initialize_Lean_Environment(lean_object*);
static bool _G_initialized = false;
lean_object* initialize_Lean_AuxRecursor(lean_object* w) {
lean_object * res;
if (_G_initialized) return lean_io_result_mk_ok(lean_box(0));
_G_initialized = true;
res = initialize_Init(lean_io_mk_world());
if (lean_io_result_is_error(res)) return res;
lean_dec_ref(res);
res = initialize_Lean_Environment(lean_io_mk_world());
if (lean_io_result_is_error(res)) return res;
lean_dec_ref(res);
l_Lean_casesOnSuffix = _init_l_Lean_casesOnSuffix();
lean_mark_persistent(l_Lean_casesOnSuffix);
l_Lean_recOnSuffix = _init_l_Lean_recOnSuffix();
lean_mark_persistent(l_Lean_recOnSuffix);
l_Lean_brecOnSuffix = _init_l_Lean_brecOnSuffix();
lean_mark_persistent(l_Lean_brecOnSuffix);
l_Lean_binductionOnSuffix = _init_l_Lean_binductionOnSuffix();
lean_mark_persistent(l_Lean_binductionOnSuffix);
res = l_Lean_initFn____x40_Lean_AuxRecursor___hyg_39_(lean_io_mk_world());
if (lean_io_result_is_error(res)) return res;
l_Lean_auxRecExt = lean_io_result_get_value(res);
lean_mark_persistent(l_Lean_auxRecExt);
lean_dec_ref(res);
res = l_Lean_initFn____x40_Lean_AuxRecursor___hyg_101_(lean_io_mk_world());
if (lean_io_result_is_error(res)) return res;
l_Lean_noConfusionExt = lean_io_result_get_value(res);
lean_mark_persistent(l_Lean_noConfusionExt);
lean_dec_ref(res);
return lean_io_result_mk_ok(lean_box(0));
}
#ifdef __cplusplus
}
#endif
