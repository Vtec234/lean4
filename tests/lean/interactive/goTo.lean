import Lean -- needed for go-to elab/syntax

structure Bar where

structure Foo where
       --v textDocument/definition
  foo₁ : Nat
  foo₂ : Nat
  bar  : Bar

def mkFoo₁ : Foo := {
--v textDocument/definition
  foo₁ := 1
--v textDocument/declaration
  foo₂ := 2
--v textDocument/typeDefinition
  bar := ⟨⟩
}

--v textDocument/definition
inductive HandWrittenStruct where
  | mk (n : Nat)

--v textDocument/declaration
def HandWrittenStruct.n := fun | mk n => n

def hws : HandWrittenStruct := {
--v textDocument/definition
  n := 3
}

            --v textDocument/declaration
def mkFoo₂ := mkFoo₁

