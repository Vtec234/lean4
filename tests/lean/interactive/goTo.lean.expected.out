{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 6, "character": 9}}
[{"targetUri": "file:///home/wjn/lean4/src/Init/Prelude.lean",
  "targetSelectionRange":
  {"start": {"line": 341, "character": 10},
   "end": {"line": 341, "character": 13}},
  "targetRange":
  {"start": {"line": 341, "character": 0},
   "end": {"line": 343, "character": 24}},
  "originSelectionRange":
  {"start": {"line": 6, "character": 9}, "end": {"line": 6, "character": 12}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 12, "character": 2}}
[{"targetUri": "file://goTo.lean",
  "targetSelectionRange":
  {"start": {"line": 6, "character": 2}, "end": {"line": 6, "character": 6}},
  "targetRange":
  {"start": {"line": 6, "character": 2}, "end": {"line": 6, "character": 6}},
  "originSelectionRange":
  {"start": {"line": 12, "character": 2}, "end": {"line": 12, "character": 6}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 14, "character": 2}}
[{"targetUri": "file://goTo.lean",
  "targetSelectionRange":
  {"start": {"line": 7, "character": 2}, "end": {"line": 7, "character": 6}},
  "targetRange":
  {"start": {"line": 7, "character": 2}, "end": {"line": 7, "character": 6}},
  "originSelectionRange":
  {"start": {"line": 14, "character": 2}, "end": {"line": 14, "character": 6}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 16, "character": 2}}
[{"targetUri": "file://goTo.lean",
  "targetSelectionRange":
  {"start": {"line": 2, "character": 10}, "end": {"line": 2, "character": 13}},
  "targetRange":
  {"start": {"line": 2, "character": 0}, "end": {"line": 2, "character": 19}},
  "originSelectionRange":
  {"start": {"line": 16, "character": 2}, "end": {"line": 16, "character": 5}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 20, "character": 2}}
[{"targetUri": "file:///home/wjn/lean4/src/Lean/Elab/Declaration.lean",
  "targetSelectionRange":
  {"start": {"line": 140, "character": 4},
   "end": {"line": 140, "character": 19}},
  "targetRange":
  {"start": {"line": 140, "character": 0},
   "end": {"line": 161, "character": 41}},
  "originSelectionRange":
  {"start": {"line": 20, "character": 0},
   "end": {"line": 21, "character": 16}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 24, "character": 2}}
[{"targetUri": "file:///home/wjn/lean4/src/Lean/Parser/Command.lean",
  "targetSelectionRange":
  {"start": {"line": 63, "character": 28},
   "end": {"line": 63, "character": 39}},
  "targetRange":
  {"start": {"line": 63, "character": 24},
   "end": {"line": 64, "character": 164}},
  "originSelectionRange":
  {"start": {"line": 24, "character": 0},
   "end": {"line": 24, "character": 42}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 28, "character": 2}}
[{"targetUri": "file://goTo.lean",
  "targetSelectionRange":
  {"start": {"line": 24, "character": 4}, "end": {"line": 24, "character": 23}},
  "targetRange":
  {"start": {"line": 24, "character": 0}, "end": {"line": 24, "character": 42}},
  "originSelectionRange":
  {"start": {"line": 28, "character": 2}, "end": {"line": 28, "character": 3}}}]
{"textDocument": {"uri": "file://goTo.lean"},
 "position": {"line": 32, "character": 14}}
[{"targetUri": "file://goTo.lean",
  "targetSelectionRange":
  {"start": {"line": 10, "character": 4}, "end": {"line": 10, "character": 10}},
  "targetRange":
  {"start": {"line": 10, "character": 0}, "end": {"line": 17, "character": 1}},
  "originSelectionRange":
  {"start": {"line": 32, "character": 14},
   "end": {"line": 32, "character": 20}}}]
